docs/content/python/guides/enforcement/resource.md

```diff
@@ -0,0 +1,115 @@
+---
+title: "Resource-level Enforcement"
+weight: 10
+any: true
+description: >
+  Learn about enforcing request-level authorization, controlling who can perform
+  which actions on which resources.
+# draft: true
+---
+
+{{% callout "Note: 0.20.0 Alpha Feature" %}}
+  This is an API provided by the alpha release of Oso 0.20.0, meaning that it is
+  not yet officially released. You may find other docs that conflict with the
+  guidance here, so proceed at your own risk! If you have any questions, don't
+  hesitate to [reach out to us on Slack](https://join-slack.osohq.com). We're
+  here to help.
+{{% /callout %}}
+
+<div class="pb-10"></div>
+
+# Resource-level Enforcement
+
+Is the current user allowed to perform a certain action on a certain resource?
+This is the central question of **"resource-level" enforcement.**
+
+Resource-level enforcement is the bread-and-butter of application authorization.
+If you only perform one type of authorization in your app, it should be
+this. **Just about every endpoint in your application should perform some kind
+of resource-level enforcement.**
+
+The method you use for resource-level authorization is called `authorize`, and
+is exposed by an Oso enforcer. The `authorize` method takes three arguments,
+`user`, `action`, and `resource`. It doesn't return anything, but potentially
+throws an error. To handle this error, see [Authorization
+Failure](#authorization-failure).
+
+<!-- You'll see this method in a lot of our guides and examples, because it's the
+simplest way to use Oso in your app. -->
+
+```python
+oso.authorize(user, "approve", expense)
+```
+
+The `authorize` method checks all of the `allow` rules in your policy, and
+ensures that there is an `allow` rule for that applies to the given user,
+action, and resource:
+
+```polar
+allow(user: User, "approve", expense: Expense) if
+    expense.org = org and
+    user.has_org_role(org, "admin");
+```
+
+Let's see an example of `authorize` from within an endpoint:
+
+```python
+def get_expense(user, expense_id):
+    expense = db.fetch(
+        "SELECT * FROM expenses WHERE id = %", expense_id)
+    oso.authorize(user, "read", expense)
+
+    # ... process request
+```
+
+As you can see from this example, it's common to have to fetch _some_ data
+before performing authorization. To perform resource-level authorization, you
+normally need to have the resource loaded!
+
+## Authorization Failure
+
+What happens when the authorization fails? That is, what if there is not an
+`allow` rule that gives the user permission to perform the action on the
+resource? In that case, the `authorize` method will raise an
+`AuthorizationError`. There are actually two types of authorization errors,
+depending on the situation.
+
+- `NotFound` errors are for situations where the user should not even know
```

Comment:
These are called `ForbiddenError` & `NotFoundError` in the API. Use those names instead here?

Also link to API docs.